diff --git a/flashcard_cycle/js/flashcard_cycle.js b/flashcard_cycle/js/flashcard_cycle.js
index 5e6df96..bf744da 100644
--- a/flashcard_cycle/js/flashcard_cycle.js
+++ b/flashcard_cycle/js/flashcard_cycle.js
@@ -61,7 +61,7 @@
 
             .find('.flashcard-cycle-buttons')
               .wrapInner('<div class="flashcard-cycle-wrapper clearfix"></div>')
-              .find('a')
+              .find('a:not(.edit))
                 .each(
                   function() {
                     if ($(this).attr('class').length > 0) {
