diff --git a/flashcard.module b/flashcard.module
index 88398a5..98e783e 100644
--- a/flashcard.module
+++ b/flashcard.module
@@ -124,11 +124,13 @@ function flashcard_field_widget_form(&$form, &$form_state, $field, $instance, $l
     '#type' => 'textfield',
     '#title' => t('Front of card'),
     '#default_value' => isset($items[$delta]['question']) ? $items[$delta]['question'] : '',
+    '#maxlength' => 255,
   );
   $element['answer'] = array(
     '#type' => 'textfield',
     '#title' => t('Back of card'),
     '#default_value' => isset($items[$delta]['answer']) ? $items[$delta]['answer'] : '',
+    '#maxlength' => 255,
   );
   return $element;
 }
@@ -150,8 +152,8 @@ function flashcard_field_formatter_view($entity_type, $entity, $field, $instance
       '#theme' => 'item_list',
       '#items' => array(
         array(
-          'data' => check_plain($item['question']),
-          'children' => array(check_plain($item['answer'])),
+          'data' => decode_entities(check_plain($item['question'])),
+          'children' => array(decode_entities(check_plain($item['answer']))),
           'class' => array('answer'),
         ),
       ),
