Index: draggableviews.rules.inc
===================================================================
--- draggableviews.rules.inc	(revision 4789)
+++ draggableviews.rules.inc	(working copy)
@@ -14,6 +14,10 @@
         'type' => 'text',
         'label' => t('view name'),
       ),
+      'display_name' => array(
+        'type' => 'text',
+        'label' => t('view display name'),
+      ),
     ),
   );
 
Index: draggableviews.module
===================================================================
--- draggableviews.module	(revision 4789)
+++ draggableviews.module	(working copy)
@@ -507,7 +507,7 @@
 
   // Trigger the event "A view has been sorted"
   if (module_exists('rules')) {
-    rules_invoke_event('draggableviews_rules_event_sorted', $view->name);
+    rules_invoke_event('draggableviews_rules_event_sorted', $view->name, $view->current_display);
   }
 
   // Redirect form.
