From 3e29b905601c12b7ad8122279d054860b18b175a Tue, 20 Mar 2012 14:30:23 -0400
From: AlexanderAllen <AlexanderAllen@351784.no-reply.drupal.org>
Date: Tue, 20 Mar 2012 14:12:03 -0400
Subject: [PATCH] Issue #1491308 by AlexanderAllen: Broadcast current_display when triggering draggableviews_rules_event_sorted events.


diff --git a/draggableviews.module b/draggableviews.module
index 21b6fe6..7277761 100644
--- a/draggableviews.module
+++ b/draggableviews.module
@@ -507,7 +507,7 @@
 
   // Trigger the event "A view has been sorted"
   if (module_exists('rules')) {
-    rules_invoke_event('draggableviews_rules_event_sorted', $view->name);
+    rules_invoke_event('draggableviews_rules_event_sorted', $view->name, $view->current_display);
   }
 
   // Redirect form.
diff --git a/draggableviews.rules.inc b/draggableviews.rules.inc
index 986c04e..c44accf 100644
--- a/draggableviews.rules.inc
+++ b/draggableviews.rules.inc
@@ -14,6 +14,10 @@
         'type' => 'text',
         'label' => t('view name'),
       ),
+      'display_name' => array(
+        'type' => 'text',
+        'label' => t('view display name'),
+      ),
     ),
   );
 